### DISCUSSION TITLE
Account password length

### DISCUSSION THREAD

**Discussion by crazydef** [community/community#78055] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi,

1) Why is my account password limited to 72 characters in length?

2) Are you storing passwords in plain text?  If so.  Stop doing that.  If not, see question 1.

**Comment by thevindu-w** [comment]
How does that limitation imply storing plaintext passwords?
Even NIST guidelines requires allowing up to 64 characters. (source: [https://pages.nist.gov/800-63-3/sp800-63b.html](https://pages.nist.gov/800-63-3/sp800-63b.html))

**Reply by crazydef** [reply]
Because it implies the password could be being stored in a fixed-length database field.

The database field should be fixed length - it should be long enough to store the calculated hash value.  But that shouldn't be apparent to the user.

The only restriction that should be placed on a password is the minimum length.  And these days that should be at least 15 characters.  (A 15 character, lowercase-only password will take about 1000 years to crack on current hardware.) But in reality you want to use as long a password as possible.

If I want to use a 2000 character password, there should be nothing stopping me.

**Reply by thevindu-w** [reply]
But then why does NIST say, "Allow at least 64 characters in length"? (from the link I posted above). If a site is not storing plaintext passwords, and if that implies there is no need to set a maximum length at all, why does NIST specify something about the minimum limit for the allowed maximum length? They could have just said, "Do not store plaintext passwords.". That means there should be other reasons for this (not only storing plaintext passwords).

Also, there are length limitations in password hashing algorithms. For example Bcrypt has maximum length limit of the input (password). (source: [security.stackexchange.com/a/184090](https://security.stackexchange.com/a/184090))
